pkg/util/cloudproviders/cloudfoundry/cccache.go

```diff
@@ -90,7 +90,7 @@ type CCCache struct {
 	segmentBySpaceGUID   map[string]*cfclient.IsolationSegment
 	segmentByOrgGUID     map[string]*cfclient.IsolationSegment
 	appsBatchSize        int
-	activeResources      map[string]chan interface{}
+	locksByResource      map[string]*sync.RWMutex
```

Comment:
nit- thoughts on making `locksByResource` a nested mapping ie you get the lock by accessing `locksByResource[resourceName][guid]`? Otherwise I would rename the variable to `resourceLocks` or something similar since it isn't specifically "by resource".

Reply:
I don't think it'll be worth it. I addressed these concerns in the last commits, let me know what you think!

Reply:
looks good!